src/rgw/rgw_admin.cc

```diff
@@ -4255,14 +4260,26 @@ int main(int argc, const char **argv)
       return EIO;
     }
 
+    cfgstore = DriverManager::create_config_store(dpp(), config_store_type);
+    if (!cfgstore) {
+      std::cerr << "Unable to initialize config store." << std::endl;
+      exit(1);
+    }
```

Comment:
this call to `create_config_store()` is duplicated with the block just above. that initializes a `std::unique_ptr<rgw::sal::ConfigStore> cfgstore` variable on main's stack. can we drop the global `cfgstore` you've added?